src/odb.c

```diff
@@ -634,6 +638,42 @@ int git_odb_add_disk_alternate(git_odb *odb, const char *path)
 	return git_odb__add_default_backends(odb, path, true, 0);
 }
 
+int git_odb__open_own(git_odb **out, git_repository *repo)
+{
+	git_odb *db;
+	git_buf objects_dir = GIT_BUF_INIT;
+	int error;
+
+	assert(out && repo);
+
+	*out = NULL;
+
+	if ((error = git_repository_item_path(&objects_dir, repo,
```

Comment:
This function got extracted because it's the interface point between them, and *normally* the place where the odb would initialize its backends' filesystem locations (hence it was more logical in `odb`). That's still in flux though.